camayoc/tests/qcs/cli/test_scans.py

```diff
@@ -113,270 +106,165 @@ def setup_sources():
-    :steps: Run ``qpc scan start --sources <source1> <source2> ...``
-    :expectedresults: The scan must completed without any error and a report
-        should be available.
+
+def test_create_scan_with_options(isolated_filesystem,
+                                  qpc_server_config, source):
+    """Perform a scan and disable an optional product.
+
+    :id: 79fadb3a-9e3c-4e84-890a-d2bd954c2869
+    :description: Perform a scan and disable an optional product.
+    :steps: Run ``qpc scan add --sources <source> --disable-optional-products
+        <optional-product>``
+    :expectedresults: The created scan matches specified options for options.
     """
     scan_name = uuid4()
+    source_name = config_sources()[0]['name']
     result = scan_add({
         'name': scan_name,
-        'sources': ' '.join([source['name'] for source in config_sources()]),
+        'sources': source_name,
+        'max-concurrency': 25,
+        'disabled-optional-products': 'jboss_eap',
+        'enabled-ext-product-search': 'jboss_fuse'
     })
     match = re.match(r'Scan "{}" was added.'.format(scan_name), result)
     assert match is not None
-    result = scan_start({
-        'name': scan_name,
-    })
-    match = re.match(r'Scan "(\d+)" started.', result)
-    assert match is not None
-    scan_job_id = match.group(1)
-    wait_for_scan(scan_job_id, timeout=1200)
-    result = scan_job({
-        'id': scan_job_id,
-    })
-    assert result['status'] == 'completed'
-    report_id = result['report_id']
-    assert report_id is not None
-    output_file = 'out.json'
-    report = report_detail({
-        'json': None,
-        'output-file': output_file,
-        'report': report_id,
-    })
-    with open(output_file) as report_data:
-        report = json.load(report_data)
-        assert report.get('sources', []) != []
+
+    source_output = source_show_output({'name': source_name})
+    source_output = json.loads(source_output)
+
+    scan_show_result = scan_show({'name': scan_name})
+    scan_show_result = json.loads(scan_show_result)
+    expected_result = {'id': scan_show_result['id'],
+                       'name': scan_name,
+                       'options': {
+                           'disabled_optional_products': {
+                               'jboss_brms': False,
+                               'jboss_eap': True,
+                               'jboss_fuse': False},
+                           'enabled_extended_product_search': {
+                               'jboss_brms': False,
+                               'jboss_eap': False,
+                               'jboss_fuse': True},
+                           'max_concurrency': 25},
+                       'scan_type': 'inspect',
+                       'sources': [{'id': source_output['id'],
+                                    'name': source_name,
+                                    'source_type': 'network'}]}
+
+    assert expected_result == scan_show_result
+
```

Comment:
If you wanted to be exhaustive you could somehow make this do more combinations by parametrizing the test ... but that gets done on API side too so may be redundant. Up to you